website/addons/googledrive/client.py

```diff
@@ -0,0 +1,151 @@
+import os
+import itertools
+
+import furl
+import requests
+
+from requests_oauthlib import OAuth2Session
+
+from framework import exceptions
+
+from website.util import api_url_for
+from website.addons.googledrive import settings
+
+
+class BaseClient(object):
+
+    def __init__(self, base):
+        self.base = base
+
+    @property
+    def default_headers(self):
+        return {}
+
+    def _make_request(self, method, segments, **kwargs):
+        query = kwargs.pop('query', {})
+        expects = kwargs.pop('expects', None)
+        throws = kwargs.pop('throws', exceptions.HTTPError)
```

Comment:
I believe you'll need to instantiate `HTTPError` here: `exceptions.HTTPError(400)`.